tests/legacy-cli/e2e/tests/commands/cache/cache-info.ts

```diff
@@ -0,0 +1,63 @@
+import { execAndWaitForOutputToMatch } from '../../../utils/process';
+import { updateJsonFile } from '../../../utils/project';
+
+export default async function () {
+  const originalCIValue = process.env['CI'];
+
+  try {
+    // Should be enabled by default for local builds.
+    process.env['CI'] = '0';
+    await configureCacheOptions();
+    await execAndWaitForOutputToMatch(
+      'ng',
+      ['cache', 'info'],
+      /Effective status on current machine: enabled/,
+    );
+
+    // Should be enabled by default for CI builds.
+    process.env['CI'] = '1';
+    await configureCacheOptions({ enabled: true });
+    await execAndWaitForOutputToMatch(
+      'ng',
+      ['cache', 'info'],
+      /Effective status on current machine: disabled/,
+    );
+
+    // Effective status should be enabled when 'environment' is set to 'all' or 'ci'.
+    await configureCacheOptions({ environment: 'all' });
+    await execAndWaitForOutputToMatch(
+      'ng',
+      ['cache', 'info'],
+      /Effective status on current machine: enabled/,
+    );
+
+    // Effective status should be enabled when 'environment' is set to 'ci' and run is in ci
+    process.env['CI'] = '1';
+    await configureCacheOptions({ environment: 'ci' });
+    await execAndWaitForOutputToMatch(
+      'ng',
+      ['cache', 'info'],
+      /Effective status on current machine: enabled/,
+    );
+
+    // Effective status should be disabled when 'enabled' is set to false
+    await configureCacheOptions({ environment: 'all', enabled: false });
+    await execAndWaitForOutputToMatch(
+      'ng',
+      ['cache', 'info'],
+      /Effective status on current machine: disabled/,
+    );
+  } finally {
+    process.env['CI'] = originalCIValue;
+  }
+}
+
+async function configureCacheOptions(cacheOptions?: {
```

Comment:
Should this clear/set the `CI` variable? I think it would be easy to forget to reset it between test cases and leak that state accidentally.